File: core/src/main/java/javabot/operations/time/DateUtils.java
package javabot.operations.time;

import java.util.Date;

import org.joda.time.DateTimeZone;
import org.joda.time.format.DateTimeFormat;
import org.joda.time.format.DateTimeFormatter;

public class DateUtils {
    private final DateTimeFormatter dateFormatter = DateTimeFormat.forPattern( "MMMM dd" );
    private final DateTimeFormatter timeFormatter = DateTimeFormat.forPattern( "h:mm aa" );

    public String getCurrentDateAtZone( String timezone ) {
        Long date = new Date( ).getTime( );
        String formattedDate = dateFormatter.withZone( DateTimeZone.forID( timezone ) ).print( date );
        String formattedTime = timeFormatter.withZone( DateTimeZone.forID( timezone ) ).print( date ).toLowerCase( );
        return formattedDate + " @ " + formattedTime;
    }
}

File: src/main/kotlin/javabot/operations/time/DateUtils.kt
package javabot.operations.time

import java.time.LocalDateTime
import java.time.ZoneId
import java.time.format.DateTimeFormatter

class DateUtils {
    private val dateFormatter = DateTimeFormatter.ofPattern("MMMM dd")
    private val timeFormatter = DateTimeFormatter.ofPattern("h:mm aa")

    fun getCurrentDateAtZone(timezone: String): String {
        val date = LocalDateTime.now()
        val formattedDate = dateFormatter.withZone(ZoneId.of(timezone)).format(date)
        val formattedTime = timeFormatter.withZone(ZoneId.of(timezone)).format(date).lowercase()
        return "$formattedDate @ $formattedTime"
    }
}
